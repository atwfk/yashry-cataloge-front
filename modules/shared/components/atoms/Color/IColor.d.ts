declare namespace IColor {
  export interface IProps {
    name: string;
    active: boolean;
    filterProductByColor: (colorName: string, isChecked: boolean) => void;
  }
}

export { IColor };
