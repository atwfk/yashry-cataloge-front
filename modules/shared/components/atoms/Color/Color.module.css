.color-parent {
  @apply flex items-center;
}

.color-label {
  @apply flex items-center cursor-pointer;
}

.color-bullet {
  @apply w-3 h-3 rounded-full mr-2;
}
