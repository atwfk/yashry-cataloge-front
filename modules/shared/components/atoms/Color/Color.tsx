import React from "react";
import type { FC, ReactElement, ChangeEvent } from "react";
import P from "../Paragraph/Paragraph";
import { IColor } from "./IColor";
import styles from "./Color.module.css";

const Color: FC<IColor.IProps> = ({
  name,
  active,
  filterProductByColor,
}): ReactElement => {
  const pColor = active ? "accent" : "dark";

  return (
    <div className={styles["color-parent"]}>
      <input
        type="checkbox"
        name={name}
        id={name}
        className="mr-2"
        onChange={(e: ChangeEvent<HTMLInputElement>) => {
          filterProductByColor(e.currentTarget.id, e.target.checked);
        }}
        checked={active}
        data-testid="checkbox"
      />
      <label htmlFor={name} className={styles["color-label"]}>
        <div
          className={styles["color-bullet"]}
          style={{ backgroundColor: `${name}` }}
        ></div>
        <P fontColor={pColor} fontSize="sm" fontWeight="normal">
          {name}
        </P>
      </label>
    </div>
  );
};

export default Color;
