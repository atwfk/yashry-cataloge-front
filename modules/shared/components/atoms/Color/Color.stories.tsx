import type { ReactElement } from "react";
import type { Story } from "@storybook/react";
import Color from "./Color";
import { IColor } from "./IColor";

const args = {
  title: "Atoms/Color",
  component: Color,
};

export default args;

const Template: Story<IColor.IProps> = (args): ReactElement => (
  <Color {...args} />
);

export const color = Template.bind({});
color.args = {
  name: "red",
  active: false,
};
