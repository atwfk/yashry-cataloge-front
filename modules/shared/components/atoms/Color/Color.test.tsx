import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Color from "./Color";

describe("Testing Color with snapshot", () => {
  it("should render inactive Color component", () => {
    const tree = renderer
      .create(
        <Color
          name="red"
          active={false}
          filterProductByColor={(): string => {
            return "test";
          }}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render active Color component", () => {
    const tree = renderer
      .create(
        <Color
          name="red"
          active={true}
          filterProductByColor={(): string => {
            return "test";
          }}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("Testing Color with RTL", () => {
  it("Expect color checkbox not to be checked when we pass active={false}", () => {
    render(
      <Color
        name="red"
        active={false}
        filterProductByColor={(): string => {
          return "test";
        }}
      />,
    );

    const checkbox = screen.getByTestId("checkbox");
    expect(checkbox).not.toBeChecked();
  });

  it("Expect color checkbox to be checked when we pass active={true}", () => {
    render(
      <Color
        name="red"
        active={true}
        filterProductByColor={(): string => {
          return "test";
        }}
      />,
    );

    const checkbox = screen.getByTestId("checkbox");
    expect(checkbox).toBeChecked();
  });

  it("Expect filterProductByColor to be called once", () => {
    const handleChange = jest.fn();
    render(
      <Color name="red" active={false} filterProductByColor={handleChange} />,
    );

    const checkbox = screen.getByTestId("checkbox");

    userEvent.click(checkbox);
    expect(handleChange).toHaveBeenCalledTimes(1);
  });
});
