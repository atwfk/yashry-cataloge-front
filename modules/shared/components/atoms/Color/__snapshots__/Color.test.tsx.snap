// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Color with snapshot should render active Color component 1`] = `
<div
  className="color-parent"
>
  <input
    checked={true}
    className="mr-2"
    data-testid="checkbox"
    id="red"
    name="red"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="color-label"
    htmlFor="red"
  >
    <div
      className="color-bullet"
      style={
        Object {
          "backgroundColor": "red",
        }
      }
    />
    <p
      className="text-sm text-accent font-normal"
    >
      red
    </p>
  </label>
</div>
`;

exports[`Testing Color with snapshot should render inactive Color component 1`] = `
<div
  className="color-parent"
>
  <input
    checked={false}
    className="mr-2"
    data-testid="checkbox"
    id="red"
    name="red"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="color-label"
    htmlFor="red"
  >
    <div
      className="color-bullet"
      style={
        Object {
          "backgroundColor": "red",
        }
      }
    />
    <p
      className="text-sm text-dark font-normal"
    >
      red
    </p>
  </label>
</div>
`;
